### PR TITLE
Adding no cookies reset option

### LINKED ISSUE #622: Add option to persist cookies across VU iterations
Currently k6 clears all cookies when a VU iteration ends, before the next one begins [1]. We should make this configurable (session cookies should probably still be discarded though).

Something along the lines of:
`k6 run --persist-cookies script.js`

In-script option:
```js
export let options = {
    persistCookies: true|false
};
```

[1] - https://github.com/loadimpact/k6/blob/master/js/runner.go#L319

### PR DESCRIPTION
**Pull request by luizbafilho**
closes #622

### CODE REVIEW

File: cmd/options.go [hunk 1]
```diff
@@ -67,6 +67,7 @@ func optionFlagSet() *pflag.FlagSet {
 	flags.String("summary-time-unit", "", "define the time unit used to display the trend stats. Possible units are: 's', 'ms' and 'us'")
 	flags.StringSlice("system-tags", lib.DefaultSystemTagList, "only include these system tags in metrics")
 	flags.StringSlice("tag", nil, "add a `tag` to be applied to all samples, as `[name]=[value]`")
+	flags.Bool("no-cookies-reset", false, "don't reset cookies after a VU iteration")
```
[the remark this answers is not in the document]
Comment (luizbafilho) [reply cmd/options.go]: Good point, I guess we should discuss someday what worth or not a CLI flag. I'll wipe this out for now.

**Comment by mercs600** [comment]
Hi guys, I have question @luizbafilho. Am I able to run k6 with this new option --no-cookies-reset now ? I'm trying build k6 from sources. But I got ERRO[0000] unknown flag: --no-cookies-reset. 
I need this option to share cookies between tests.

**Comment by na--** [comment]
Hey @mercs600, we didn't add a command-line option for the new setting because we didn't want to overcrowd the CLI interface with an option that probably won't be used very often. If there's enough user demand we can always enable it later, but for now you can still enable the "no cookies reset" functionality it by setting the environment variable `K6_NO_COOKIES_RESET` to `true` like this:
```
export K6_NO_COOKIES_RESET=true 
k6 run myscript.js
```
or by using the in-script option `noCookiesReset` like this:
```js
export let options = {
    noCookiesReset: true
};
```

Sorry for the confusion, I've expanded the [upcoming release notes](https://github.com/loadimpact/k6/blob/master/release%20notes/upcoming.md#new-option-no-cookies-reset-729) to mention this and later I'll also update the readme.io docs with information about it.

**Comment by mercs600** [comment]
@na-- thank you. Could you tell me something more ?  I'm trying setup cookies in setup() lifecycle. For example I make login action in setup() and there after submitForm with login credentials server returns cookies. I would like to use this cookies in default cycle when tests are executing. Is it possible ? Because right now when I setup it manually by export options I can't do this.

**Comment by na--** [comment]
Hmm not sure if I correctly understand what you're trying to do, but keep in mind that `setup()` runs in its own separate JS runtime, and each VU does as well. So you should be able to make a login/signup/etc. http request in `setup()`, get the cookies from its response and `return` them from setup so that each VU could access them. Then each VU can just set them in the [cookie jar](https://docs.k6.io/docs/cookies#section-setting-simple-cookies) at the beginning of each iteration, you wouldn't even need the `noCookiesReset` option that way. 

In any case, these types of questions and discussions are more [suited](https://github.com/loadimpact/k6/#need-help-or-want-to-contribute) to the `#support` channel in the slack chat or for stackoverflow, not closed github issues :smile:

**Comment by mercs600** [comment]
I resolved my issue by getting cookie from previous response in setup() cycle.
Thank you for your help and sorry for reopen ;-)

**Pull request merged**